### ISSUE TITLE
non-linear progression

### ISSUE THREAD

**Issue by M1cha** [M1cha/hyprctl-custom#1]
Also just want to let you know that hyprctl-custom works great with fusuma:
```
swipe:
  3:
    up:
      update:
        command: "~/.local/bin/hyprctl-custom cursor-zoom in"
    down:
      update:
        command: "~/.local/bin/hyprctl-custom cursor-zoom out"
interval:
  swipe: 0.2
```

I mostly use keyboard and trying to migrate to mouseless (touchpadless) life, but I appreciate the gestures when I am too lazy or just using laptop one handed.

The only thing I am missing is non-linear progression, some curve. Because the more you are zoomed in, the smaller next "step" feels. Hope you understood what I mean. But this already involves some config file I guess, which is not that easy, at least for me. Just my thoughts :)

_Originally posted by @ZerdoX-x in https://github.com/M1cha/dotfiles/issues/1#issuecomment-1720258881_

**Comment by M1cha** [comment]
> The only thing I am missing is non-linear progression, some curve. Because the more you are zoomed in, the smaller next "step" feels. Hope you understood what I mean. But this already involves some config file I guess, which is not that easy, at least for me. Just my thoughts :)

I don't share that experience. Are sure that's not an artifact of how often your gesture triggers the command? Either way, implementing this would be easy, but I need to know what kind of curve you want.

**Comment by ekhodx** [comment]
Thank you for raising this up.

> Are sure that's not an artifact of how often your gesture triggers the command?

Of course, I can feel it using both touchpad gesture and regular hypr bind. I will try to show you as soon https://github.com/hyprwm/xdg-desktop-portal-hyprland/issues/97 gets resolved because I don't have ability to record screen yet

**Comment by M1cha** [comment]
I think I get what you mean now. I'll do some research and experiment with some curves.

**Comment by M1cha** [comment]
I think the issue is that we expect the zoom factor to be applied to what we see on the screen, not to the whole thing. So I've implemented scaling the zoom factor by the current zoom factor. To me that feels much better now. Here's a binary you can use to test: https://github.com/M1cha/hyprctl-custom/suites/16275097005/artifacts/929155905.
Let me know what you think.

**Comment by ekhodx** [comment]
Oh, yes, this is probably better and more consistent then what I proposed. Also I think "progressively slow zoom" problem is more noticeable on larger screens, mine is 2560x1600.

> To me that feels much better now

Nice, glad to hear

> Let me know what you think.

I tested and it does work and fix described issue, this one is much better. Are you going to merge this in?

> Here's a binary

I am on arm64, so I just compiled `relative-cursor-zoom` branch myself :)

UPD: Issue can be closed, thank you
